ambry-tools/src/main/java/com.github.ambry/tools/admin/ServerAdminTool.java

```diff
@@ -327,18 +351,46 @@ private static void writeBufferToFile(ByteBuffer buffer, FileOutputStream output
    * @throws IOException
    * @throws TimeoutException
    */
-  private static void sendControlRequest(ServerAdminTool serverAdminTool, ClusterMap clusterMap, DataNodeId dataNodeId,
-      String partitionId, RequestOrResponseType toControl, boolean enable) throws IOException, TimeoutException {
+  private static void sendRequestControlRequest(ServerAdminTool serverAdminTool, ClusterMap clusterMap,
+      DataNodeId dataNodeId, String partitionId, RequestOrResponseType toControl, boolean enable)
+      throws IOException, TimeoutException {
     ServerErrorCode errorCode = serverAdminTool.controlRequest(dataNodeId, partitionId, toControl, enable, clusterMap);
     if (errorCode == ServerErrorCode.No_Error) {
       LOGGER.info("{} enable state has been set to {} for {} on {}", toControl, enable, partitionId, dataNodeId);
     } else {
-      LOGGER.error("From {}, received server error code {} for enable state {} request for {} on {}", dataNodeId,
+      LOGGER.error("From {}, received server error code {} for request to set enable state {} for {} on {}", dataNodeId,
           errorCode, enable, toControl, partitionId);
     }
   }
 
   /**
+   * Sends a {@link ReplicationControlAdminRequest} to {@code dataNodeId} to set enable status of replication from
+   * {@code origins} to {@code enable} for {@code partitionId}.
+   * @param serverAdminTool the {@link ServerAdminTool} instance to use.
+   * @param clusterMap the {@link ClusterMap} to use.
+   * @param dataNodeId the {@link DataNodeId} to send the request to.
+   * @param partitionId the partition id (string) on which the operation will take place. Can be {@code null}.
+   * @param origins the names of the datacenters from which replication should be controlled.
+   * @param enable the enable (or disable) status required for {@code toControl}.
+   * @throws IOException
+   * @throws TimeoutException
+   */
+  private static void sendReplicationControlRequest(ServerAdminTool serverAdminTool, ClusterMap clusterMap,
+      DataNodeId dataNodeId, String partitionId, List<String> origins, boolean enable)
+      throws IOException, TimeoutException {
+    ServerErrorCode errorCode =
+        serverAdminTool.controlReplication(dataNodeId, partitionId, origins, enable, clusterMap);
+    if (errorCode == ServerErrorCode.No_Error) {
+      LOGGER.info("Enable state of replication from {} has been set to {} for {} on {}", origins, enable, partitionId,
```

Comment:
I think logger provides the flexibility to either redirect to stdout or file not to mention different levels of logging.